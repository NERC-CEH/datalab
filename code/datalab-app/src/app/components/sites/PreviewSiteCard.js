import React, { Component } from 'react';
import { connect } from 'react-redux';
import StackCard from '../stacks/StackCard';

class PreviewSiteCard extends Component {
  render() {
    return (
      <StackCard stack={this.props.stack} />
    );
  }
}

function mapStateToProps({ form }) {
  let stack = {};
  if (form && form.createSite && form.createSite.values) {
    stack = form.createSite.values;
  }
  return {
    stack,
  };
}

export { PreviewSiteCard as PurePreviewSiteCard }; // export for testing
export default connect(mapStateToProps)(PreviewSiteCard);
