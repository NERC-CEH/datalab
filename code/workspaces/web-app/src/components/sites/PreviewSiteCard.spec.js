import React from 'react';
import { shallow } from 'enzyme';
import createStore from 'redux-mock-store';
import PreviewSiteCard, { PurePreviewSiteCard } from './PreviewSiteCard';

const stack = { name: 'Name' };

describe('Preview SiteCard', () => {
  function createDefaultStore() {
    return createStore()({
      modalDialog: { open: false },
      form: { createSite: { values: stack } },
    });
  }

  describe('is a connected component which', () => {
    function shallowRenderConnected(store) {
      const props = {
        store,
        PrivateComponent: () => {},
        PublicComponent: () => {},
      };

      return shallow(<PreviewSiteCard {...props} />).find('PreviewSiteCard');
    }

    it('extracts the form values from the redux state', () => {
      const store = createStore()({
        modalDialog: { open: false },
      });

      const output = shallowRenderConnected(store);

      expect(output.prop('stack')).toEqual({});
    });

    it('provides empty stack if the form does not yet exist in the redux state', () => {
      const store = createDefaultStore();

      const output = shallowRenderConnected(store);

      expect(output.prop('stack')).toEqual(stack);
    });
  });

  describe('is a component which', () => {
    function shallowRenderPure() {
      const props = {
        stack: 'expectedStack',
      };

      return shallow(<PurePreviewSiteCard {...props} />);
    }
    it('passes correct props to StackCards', () => {
      expect(shallowRenderPure()).toMatchSnapshot();
    });
  });
});
