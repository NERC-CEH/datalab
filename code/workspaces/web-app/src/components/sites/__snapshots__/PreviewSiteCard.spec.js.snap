// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview SiteCard is a component which passes correct props to StackCards 1`] = `
<WithStyles(StackCard)
  deletePermission="delete"
  openPermission="open"
  stack="expectedStack"
  typeName="site"
  userPermissions={
    Array [
      "open",
      "delete",
    ]
  }
/>
`;
